internal/k8s/validation_test.go

```diff
@@ -1411,20 +1531,87 @@ func TestValidateNginxIngressAnnotations(t *testing.T) {
 			specServices:          map[string]bool{},
 			isPlus:                true,
 			appProtectEnabled:     true,
+			appProtectDosEnabled:  false,
 			internalRoutesEnabled: false,
 			expectedErrors: []string{
 				`annotations.appprotect.f5.com/app-protect-security-log-enable: Invalid value: "not_a_boolean": must be a boolean`,
 			},
 			msg: "invalid appprotect.f5.com/app-protect-security-log-enable annotation",
 		},
-
+		{
+			annotations: map[string]string{
+				"appprotectdos.f5.com/app-protect-dos-resource": "dos-resource-name",
+			},
+			specServices:          map[string]bool{},
+			isPlus:                true,
+			appProtectEnabled:     false,
+			appProtectDosEnabled:  false,
+			internalRoutesEnabled: false,
+			expectedErrors: []string{
+				"annotations.appprotectdos.f5.com/app-protect-dos-resource: Forbidden: annotation requires AppProtectDos",
+			},
+			msg: "invalid appprotectdos.f5.com/app-protect-dos-resource annotation, requires app protect dos",
+		},
+		{
+			annotations: map[string]string{
+				"appprotectdos.f5.com/app-protect-dos-resource": "dos-resource-name",
+			},
+			specServices:          map[string]bool{},
+			isPlus:                true,
+			appProtectEnabled:     false,
+			appProtectDosEnabled:  true,
+			internalRoutesEnabled: false,
+			expectedErrors:        nil,
+			msg:                   "valid appprotectdos.f5.com/app-protect-dos-enable annotation",
```

Comment:
better for case msg to be unique, because that's how we can identify the failed cases

Reply:
good point, I've updated all the test case msgs to be more descriptive.